code/modules/vehicles/mecha/equipment/tools/work_tools.dm

```diff
@@ -37,31 +37,35 @@
 		return
 	if(!workmech.cargo_hold)
 		CRASH("Mech [chassis] has a clamp device, but no internal storage. This should be impossible.")
+
 	if(ismecha(target))
 		var/obj/vehicle/sealed/mecha/M = target
 		var/have_ammo
 		for(var/obj/item/mecha_ammo/box in workmech.cargo_hold.contents)
 			if(istype(box, /obj/item/mecha_ammo) && box.rounds)
 				have_ammo = TRUE
 				if(M.ammo_resupply(box, source, TRUE))
-					return
+					return ..()
 		if(have_ammo)
 			to_chat(source, "No further supplies can be provided to [M].")
 		else
 			to_chat(source, "No providable supplies found in cargo hold")
+		return
 
-	else if(isobj(target))
+	if(istype(target, /obj/machinery/door/firedoor))
+		var/obj/machinery/door/firedoor/targetfiredoor = target
+		playsound(chassis, clampsound, 50, FALSE, -6)
+		targetfiredoor.try_to_crowbar(src, source)
+		return ..()
+
+	if(istype(target, /obj/machinery/door/airlock))
+		var/obj/machinery/door/airlock/targetairlock = target
+		playsound(chassis, clampsound, 50, FALSE, -6)
+		targetairlock.try_to_crowbar(src, source, TRUE)
+		return ..()
```

Comment:
Annoying that `try_to_crowbar` is on `/door/` but you still need to typecast for two types of doors because you don't want to feed back when attempting to pry a door you cannot pry.